pkg/generate/app/componentresolvers.go

```diff
@@ -38,51 +41,47 @@ type PerfectMatchWeightedResolver []WeightedResolver
 
 // Resolve resolves the provided input and returns only exact matches
 func (r PerfectMatchWeightedResolver) Resolve(value string) (*ComponentMatch, error) {
-	imperfect := ScoredComponentMatches{}
-	var group WeightedResolvers
+	var errs []error
+	candidates := ScoredComponentMatches{}
+	var group MultiSimpleSearcher
+	var groupWeight float32 = 0.0
 	for i, resolver := range r {
 		// lump all resolvers with the same weight into a single group
-		if len(group) == 0 || resolver.Weight == group[0].Weight {
-			group = append(group, resolver)
-			if i != len(r)-1 && r[i+1].Weight == group[0].Weight {
+		if len(group) == 0 || resolver.Weight == groupWeight {
+			group = append(group, resolver.Searcher)
+			groupWeight = resolver.Weight
+			if i != len(r)-1 && r[i+1].Weight == groupWeight {
 				continue
 			}
 		}
-		exact, inexact, err := resolveExact(group, value)
-		switch {
-		case err != nil:
+		matches, err := group.Search(true, value)
+		if err != nil {
 			glog.V(5).Infof("Error from resolver: %v\n", err)
-			return nil, err
-		case exact != nil:
-			if exact.Score == 0.0 {
-				return exact, nil
-			}
+			errs = append(errs, err...)
+		}
+
+		sort.Sort(ScoredComponentMatches(matches))
+		if len(matches) > 0 && matches[0].Score == 0.0 && (len(matches) == 1 || matches[1].Score != 0.0) {
+			return matches[0], errors.NewAggregate(errs)
+		}
+		for _, m := range matches {
 			if resolver.Weight != 0.0 {
-				exact.Score = resolver.Weight * exact.Score
-			}
-			imperfect = append(imperfect, exact)
-		case len(inexact) > 0:
-			sort.Sort(ScoredComponentMatches(inexact))
-			if inexact[0].Score == 0.0 && (len(inexact) == 1 || inexact[1].Score != 0.0) {
-				return inexact[0], nil
-			}
-			for _, m := range inexact {
-				if resolver.Weight != 0.0 {
-					m.Score = resolver.Weight * m.Score
-				}
-				imperfect = append(imperfect, m)
+				m.Score = resolver.Weight * m.Score
```

Comment:
though it doesn't really matter, this would be more readable if you used groupWeight instead of resolver.Weight here and above.